src/GovUk.Education.ExploreEducationStatistics.Admin/Services/PublicationReleaseOrderService.cs

```diff
@@ -0,0 +1,226 @@
+using GovUk.Education.ExploreEducationStatistics.Admin.Services.Interfaces;
+using GovUk.Education.ExploreEducationStatistics.Admin.ViewModels;
+using GovUk.Education.ExploreEducationStatistics.Content.Model;
+using GovUk.Education.ExploreEducationStatistics.Content.Model.Database;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Release = GovUk.Education.ExploreEducationStatistics.Content.Model.Release;
+
+namespace GovUk.Education.ExploreEducationStatistics.Admin.Services
+{
+    public class PublicationReleaseOrderService : IPublicationReleaseOrderService
+    {
+        private readonly ContentDbContext _context;
+
+        public PublicationReleaseOrderService(
+            ContentDbContext context)
+        {
+            _context = context;
+        }
+
+        public async Task CreateForCreateLegacyRelease(Guid publicationId, Guid releaseId)
+        {
+            var publication = await GetPublication(publicationId);
+
+            if (releaseId == Guid.Empty)
+            {
+                throw new ArgumentNullException(nameof(releaseId));
+            }
+
+            publication.ReleaseOrders.Add(new()
+            {
+                ReleaseId = releaseId,
+                IsLegacy = true,
+                IsDraft = false,
+                Order = publication.ReleaseOrders.Count + 1 // Add to the top of the list rather than presume the intended position
+            });
+
+            _context.Update(publication);
+        }
+
+        public async Task DeleteForDeleteLegacyRelease(Guid publicationId, Guid releaseId)
+        {
+            var publication = await GetPublication(publicationId);
+
+            if (releaseId == Guid.Empty)
+            {
+                throw new ArgumentNullException(nameof(releaseId));
+            }
+
+            var legacyRelease = await _context.LegacyReleases.FindAsync(releaseId)
+                ?? throw new KeyNotFoundException($"No matching {nameof(LegacyRelease)} found with ID {releaseId}");
```

Comment:
I would replace this with

```csharp
var legacyRelease = await _context.LegacyReleases.FirstAsync(lr => lr.Id == legacyReleaseId);
```

It will throw an exception if it's not found which saves you having to handle it in this custom way.

It would be worth making this change elsewhere too as I can see this style is used quite a lot.

Reply:
I opted for the more descriptive throws to help identify an exact cause if any errors arise which need to be investigated. These methods manage different entities and rely on several keys, so it might not be immediately obvious (from a log message) when a generic message is returned.

I can simplify to use the normal approach if preferred, but just thought I should point out the intention before I undo it.

Reply:
Cool, I see that, thanks. Maybe keep some of them in then where you are finding items within `ReleaseOrders` if that's going to help identify problems.

I don't see much of a need for it when getting resources from the database like Publication, Release, or in the example above, LegacyRelease. We don't do anything like that elsewhere in the codebase.

Personally I prefer to keep the code concise which is easier to read, and rely on tests for things like this. We can identify problems from line numbers in stack traces too.